staging/src/k8s.io/client-go/applyconfigurations/admissionregistration/v1/rule.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by applyconfiguration-gen. DO NOT EDIT.
+
+package v1
+
+import (
+	v1 "k8s.io/api/admissionregistration/v1"
+)
+
+// RuleApplyConfiguration represents an declarative configuration of the Rule type for use
+// with apply.
+type RuleApplyConfiguration struct {
+	APIGroups   []string      `json:"apiGroups,omitempty"`
+	APIVersions []string      `json:"apiVersions,omitempty"`
+	Resources   []string      `json:"resources,omitempty"`
+	Scope       *v1.ScopeType `json:"scope,omitempty"`
+}
+
+// RuleApplyConfiguration constructs an declarative configuration of the Rule type for use with
+// apply.
+func Rule() *RuleApplyConfiguration {
+	return &RuleApplyConfiguration{}
+}
+
+// WithAPIGroups sets the APIGroups field in the declarative configuration to the given value
+// and returns the receiver, so that objects can be build by chaining "With" function invocations.
+func (b *RuleApplyConfiguration) WithAPIGroups(value []string) *RuleApplyConfiguration {
```

Comment:
Taking `(values ...string)` might make for less verbose invocation. What do you think?

Reply:
Yeah, I'll make all slices varargs. I can't remember now why we switched back to a slice, but I'll fix.